crates/ra_db/src/input.rs

```diff
@@ -312,12 +312,12 @@ impl fmt::Display for Edition {
 }
 
 impl Env {
-    pub fn set(&mut self, env: &str, value: String) {
-        self.entries.insert(env.to_owned(), value);
+    pub fn set<K: AsRef<str>, V: AsRef<str>>(&mut self, env: K, value: V) {
+        self.entries.insert(env.as_ref().to_owned(), value.as_ref().to_owned());
     }
 
-    pub fn get(&self, env: &str) -> Option<String> {
-        self.entries.get(env).cloned()
+    pub fn get<K: AsRef<str>>(&self, env: K) -> Option<String> {
```

Comment:
Similarly, I'd keep this as `&str`, but, if we are going to be generic, we need a `Borrow` here (rule of thumb: when forwarding/delegating something, use the same trait bounds)